src/List.js

```diff
@@ -24,14 +22,24 @@ const List = () => {
     itemRef.update(newData);
   };
 
+  const handleSearch = e => {
+    setTerm(e.target.value);
+    const filterItems = items.filter(item => {
+      return item.name.includes(e.target.value);
+    });
+    // console.log(filterItems);
+    setfilteredItems(filterItems);
+  };
+
   return (
     <div>
       <h1>Items</h1>
-      {emptyList ? (
+      <input type="search" value={searchTerm} onChange={handleSearch}></input>
+      {filteredItems.length === 0 ? (
         <span>Your list is empty! Please add an item.</span>
```

Comment:
Just from a UX/UI perspective, when a user filters through their list and the item doesn't exist, we should maybe have a different message like 'item not found' or 'item does not exist, please add to your list'. I still think we should have the 'your list is empty, please add an item' for when the list is completely empty.

This isn't a critical thing, but maybe a stretch goal for the future.

Reply:
Absolutely, Let’s have a different message for the case where there are no items in the list and the case where no items match the search term